### PR TITLE
adding .kitchen.yml file, adding placeholder for test platforms

### PR DESCRIPTION
**Pull request by EslamElHusseiny**

### CODE REVIEW

File: kitchenci/.kitchen.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  name: docker
+
+provisioner:
+  name: puppet_apply
+  puppet_version: "3.8.1"
+  mainfests_path: manifests
+  modules_path: modules
+
+platforms:
+  {platform-placeholder}
```
Comment (anirvanc) [inline kitchenci/.kitchen.yml]: How are you thinking of using this?

File: kitchenci/.kitchen.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+driver:
+  name: docker
+
+provisioner:
+  name: puppet_apply
+  puppet_version: "3.8.1"
```
Comment (anirvanc) [inline kitchenci/.kitchen.yml]: Might be worth thinking of parameterising this as well. What do you think?

File: kitchenci/.kitchen.yml (hunk 1 above)
Comment (EslamElHusseiny) [inline kitchenci/.kitchen.yml]: well the distribution parameter in jenkins job will be a text with a pre-defined syntax e.x.

-name: centos-6.6
-name: ubuntu-14.04
and replacing the {platform-placeholder} with the value of distribution parameter in .kitchen.yml file

File: kitchenci/.kitchen.yml (hunk 2 above)
Comment (EslamElHusseiny) [inline kitchenci/.kitchen.yml]: the puppet_version is already parametrized, I just wrote the latest available version as a default value, 
for driver value, yeah it worth parametrizing it, but this should be done from jenkins job configuration side with docker as a default value in .kitchen.yml file, make sense ?

File: kitchenci/.kitchen.yml (hunk 1 above)
Comment (anirvanc) [inline kitchenci/.kitchen.yml]: Ok. But, I still can't see how `{platform-placeholder}` would be converted to `-name: centos-6.6`. Can you point that out to me?

File: kitchenci/.kitchen.yml (hunk 2 above)
Comment (anirvanc) [inline kitchenci/.kitchen.yml]: Ok, I get that. but should it be better to a parameter similar to the one you have used for platforms place holder?

Comment (EslamElHusseiny) [inline kitchenci/.kitchen.yml]: Okay, I will modify it

File: kitchenci/.kitchen.yml (hunk 1 above)
Comment (EslamElHusseiny) [inline kitchenci/.kitchen.yml]: actually in jenkins job itself there is a step that substitute the {platform-placeholder} with value of ${Distribution} parameter as following 

sed -i -e "s/{platform-placeholder}/${Distribution}/" .kitchen.yml

File: kitchenci/Gemfile [hunk 3]
```diff
@@ -5,3 +5,5 @@ gem "test-kitchen"
 gem "kitchen-puppet"
 gem "kitchen-docker"
 gem "librarian-puppet"
+gem "kitchen-vagrant"
+gem "librarian-puppet"
```
Comment (anirvanc) [inline kitchenci/Gemfile]: This seems to be a duplicate inclusion of the same gem as on `line 7`.

Comment (EslamElHusseiny) [inline kitchenci/Gemfile]: sorry, fixed.

**Comment by anirvanc** [comment]
LGTM :+1:

**Pull request merged**